README.md

```diff
@@ -114,12 +115,70 @@ struct Contact {
 }
 ```
 
+## Custom JSON*able Implementation ##
+
+If you want to nest some types that are not currently supported (for example: enums with associated values) or if you want to provide a special implementation of the serde methods for those types, you can simply implement the `JSONSerializable` and/or `JSONDeserializable` protocols.
+
+For example, if `Job` is an enum, you can still add it to `Contact`:
+
+``` swift
+struct Contact {
+    let id: String
+    let job: Job
+}
+
+enum Job: JSONSerializable {
+    […]
+
+    func toJSONObject() -> [String: Any] {
+        // Implemnt your custom serializer.
+    }
+}
+```
+
 # Installation #
 
-The current installation process is pretty basic and does not support any dependency managers yet.  
+## With CocoaPods ##
+
+Add `pod 'Sourcery-AutoJSONSerializable'` to your [Podfile](https://guides.cocoapods.org/using/using-cocoapods.html).
+
+Configure your `.sourcery.yml` file with this:
+
+``` yaml
+[…]
+templates:
+  - Pods/Sourcery-AutoJSONSerializable/Templates/
+  […]
+```
+
+Add those protocoles anywhere in your source code:
+
+``` swift
+protocol AutoJSONSerializable {}
```

Comment:
Just a side note here, I'm thinking of removing the need of these protocols altogether and use a Sourcery annotations instead:
```swift
// sourcery: AutoJSONSerializable
struct Contact {
    [...]
}
```